### ISSUE TITLE
[Data No BlaBla: Introduction à Tensorflow 2.0] Florient Chouteau

### ISSUE THREAD

**Issue by fchouteau** [TlseDataScience/meetup-cfp#23]
## Le speaker
Florient Chouteau, Ingénieur Deep Learning à Airbus Defence and Space

* Twitter handle: @foxchouteau

## Sujet de votre conférence
Data No BlaBla: Introduction à Tensorflow 2.0

## Description de votre conférence

Tensorflow 2.0 sort bientôt !

Le mastodonde du Deep Learning par Google fait peau neuve en 2019 avec l'annonce de tensorflow 2.0, pour l'instant disponible en alpha.

Qu'est-ce qui change ? Tout. Ou presque: tf 2.0 propose surtout un changement de paradigme dans la façon de penser, et de rédiger. 

"TensorFlow 2.0 executes eagerly (like Python normally does) and in 2.0, graphs and sessions should feel like implementation details". En éxagérant un peu (mais à peine), cette phrase seule pourrait amorcer la réconciliation avec une communauté qui lorgne depuis quelques mois sur la jolie alternative de Facebook, avec lequel les utilisateurs trustent les podiums Kaggle et avancent encore et toujours l'état de l'art. 

Terminé l'écriture de graphes statiques, terminé les sess.run(), place - à nouveau - à une écriture plus simple, plus pythonique. Tout en gardant la possibilité d'obtenir des graphes performants avec `Autograph` (`tf.function()`)

L'autre point, tout aussi (voir plus) intéressant pour la pratique de tous les jours, est la poursuite de la refonte des APIs, avec le passage de tf.keras comme le point de référence "pour tous les niveaux d'abstraction", avec pour objectif de pouvoir y intégrer du code personnalité (et donc bas niveau).

Alors, pourquoi ne pas découvrir ensemble de quoi il en retourne ?

**Voici donc une proposition de DataNoBlabla ayant pour objectif d'explorer très brièvement tensorflow 2.0.**

A partir d'un jeu de données simple, composé de l'animal mignon favori de la communauté, et de quelques notebooks jupyter, l'idée est de parcourir les différentes API (majoritairement haut niveau) de Tensorflow, en rentrant progressivement dans les détails:

- tf.dataset pour charger et pré-traiter ses données (et les explorer en même temps !)
- tf.keras pour écrire un premier modèle et l'entraîner
- La sauvegarde et le chargement d'un modèle, ainsi que sa réutilisation
- Une introduction aux opérations personnalisées (layer personnalisé, `tf.function()`)
- L'écriture d'une boucle d'entraînement personnalisée

Le problème abordé sera la classification d'image, avec un twist pour rendre les choses un peu plus intéressantes ! 

### Côté organisateur

Un projet Google Cloud Platform avec n instances Google Cloud ML Notebook, les quotas GPU nécessaires, sur lequel on peut ajouter les adresses mails des participants en éditeur pour la durée du TP. Cela permettra aux participants de se connecter au serveur jupyter de l'instance sans avoir besoin d'un tunnel ssh.

Au vu du jeu de données, on pourra partager une machine (voire une même carte graphique) pour plusieurs participants

Au vu de la durée du TP, les instances pourront être préemptibles et donc ne devraient pas couter bien chères. 

### Côté participant

- Un laptop avec un navigateur internet. La possibilité d'établir une connexion ssh est bienvenue, mais ce n'est pas strictement nécessaire.

## Informations diverses

* Thématique, Labels : DataNoBlabla, Deep Learning, Tensorflow, Image
* Niveau de difficulté (débutant|avancé|confirmé): débutant (avoir les bases en python, savoir à peu près comment fonctionne le deep learning)
* Durée (quickie 15-30min, conférence >1h) : Data NoBlabla, environ 3h
* Format (slides, live-coding, les deux): TP. Notebooks à trous
* Dispo ou indispo (après mars, dès que possible, pas entre X et Y): 
Avril / Mai / Juin
Par exemple vendredi soir & samedi matin

**Comment by fchouteau** [comment]
Plus d'actualité pour l'instant
